### PR TITLE
More efficient group provider initialization

### PR DESCRIPTION
**Pull request by alsamylkin**
1. We are using a really inefficient way to reverse a slice.
2. WithProvider can't be used outside of the `config` package, because `providerGroup` is private.

### CODE REVIEW

File: modules/task/cherami/cherami_test.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func TestBackendWorkflowWorkerPanic(t *testing.T) {
 	}
 	// Publish valid message
 	publish(t, m, bknd, deliveryCh, nil, nil)
-	time.Sleep(30 * time.Millisecond)
+	time.Sleep(100 * time.Millisecond)
```
Comment (alsamylkin) [inline modules/task/cherami/cherami_test.go]: https://travis-ci.org/uber-go/fx/jobs/223243446

Comment (glibsm) [inline modules/task/cherami/cherami_test.go]: what's this got to do with group init of providers, though? :(

File: config/provider_group.go [hunk 2]
```diff
@@ -25,22 +25,18 @@ type providerGroup struct {
...
 }
 
-// NewProviderGroup creates a configuration provider from a group of backends
+// NewProviderGroup creates a configuration provider from a group of backends.
+// The highest priority provider is the last.
 func NewProviderGroup(name string, providers ...Provider) Provider {
-	group := providerGroup{
-		name: name,
+	l := len(providers)
+	for i := 0; i < l/2; i++ {
```
Comment (glibsm) [inline config/provider_group.go]: stared at this for way to long to understand what's going on

File: modules/task/cherami/cherami_test.go (hunk 1 above)
Comment (alsamylkin) [inline modules/task/cherami/cherami_test.go]: Nothing, sleep is flaky as predicted.

**Pull request merged**